From 3bda554addcf4f1640b5bae658ca178668401664 Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
Date: Tue, 2 Jun 2020 12:47:42 +0200
Subject: [PATCH] kabi: ppc64le: prevent struct dma_map_ops to become defined

Patch-mainline: Never, kabi
References: jsc#SLE-12423

Prevent following kabi breakage on ppc64le:

KABI: symbol of_fdt_unflatten_tree(mod:vmlinux) changed crc from 0x1ad40ab5 to 0x810f062c
KABI: aborting due to kabi changes.
Detailed kabi warnings:
drivers/of/fdt.c:569: warning: of_fdt_unflatten_tree: modversion changed because of changes in struct dma_map_ops (became defined)

Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/of/fdt.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/of/fdt.c b/drivers/of/fdt.c
index 03bb386c3..d492ccdc5 100644
--- a/drivers/of/fdt.c
+++ b/drivers/of/fdt.c
@@ -27,7 +27,9 @@
 #include <linux/debugfs.h>
 #include <linux/serial_core.h>
 #include <linux/sysfs.h>
+#ifndef __GENKSYMS__
 #include <linux/random.h>
+#endif
 
 #include <asm/setup.h>  /* for COMMAND_LINE_SIZE */
 #include <asm/page.h>
-- 
2.16.4

